From e8ad571e2a1e25831a4f2b294146c7dd7e022aa6 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Ozawa <ozawa@apache.org>
Date: Fri, 13 Feb 2015 02:27:54 +0900
Subject: [PATCH 1396/2848] MAPREDUCE-6221. Stringifier is left unclosed in
 Chain#getChainElementConf(). Contributed by Ted
 Yu.

(cherry picked from commit 9b0ba59b8284fae132535fbca5ce372d7a6c38c0)
(cherry picked from commit 9d57392f5b0b6b129c3291f94f0a1f75479c8a67)

Change-Id: I663ab65c05ef50aebb97a1ff98d7c8c437f607aa
---
 .../apache/hadoop/mapreduce/lib/chain/Chain.java   |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/chain/Chain.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/chain/Chain.java
index 384524a..208616b 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/chain/Chain.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/chain/Chain.java
@@ -575,9 +575,8 @@ protected static int getIndex(Configuration conf, String prefix) {
   protected static Configuration getChainElementConf(Configuration jobConf,
       String confKey) {
     Configuration conf = null;
-    try {
-      Stringifier<Configuration> stringifier = 
-        new DefaultStringifier<Configuration>(jobConf, Configuration.class);
+    try (Stringifier<Configuration> stringifier =
+        new DefaultStringifier<Configuration>(jobConf, Configuration.class);) {
       String confString = jobConf.get(confKey, null);
       if (confString != null) {
         conf = stringifier.fromString(jobConf.get(confKey, null));
-- 
1.7.9.5

